src/Rule/Length.php

```diff
@@ -62,7 +62,7 @@ public function __construct($length)
      */
     public function validate($value)
     {
-        $actualLength = strlen($value);
+        $actualLength = mb_strlen($value, mb_detect_encoding($value));
```

Comment:
I think its dangerous to use `mb_detect_encoding` here, as it could return false and would case `mb_strlen` to return false as well.

Reply:
Why not use an if else statement for this ?

Reply:
@harikt That would do the trick in this case. Yet I would rather like to see something like `$validator->setEncoding()` support too, that feels like a more solid solution than trying to detect encoding. In there you can still put `mb_detect_encoding` if you like, but than you're aware that it's happening. What do you think?

Reply:
Also, I see another PR is currently open which adds an extra parameter for encoding to every function, that combined with what I stated above feels like an ideal solution to me.
https://github.com/particle-php/Validator/pull/161

Reply:
Good. I was also looking at how this is solved at https://github.com/auraphp/Aura.Filter/blob/ac33b700379d1cb978a077328425153060a91ab5/src/Rule/AbstractStrlen.php#L57 . Why is the PR #161 waiting so long then ? PR on 2016 , now 2017 . Any reason it is waiting so long ?

Reply:
Because we've been slacking. I'll review the lost of open PRs this weekend and create a new release. Thanks for the input.

Reply:
@berry-langerak I noticed a few other PR's also opened on other repos State . May be you want to look those also ? Just pointing in case you have forgoten. Don't take it wrong. Love progress of OSS .